### ISSUE TITLE
A way to exclude certain stylesheets...

### ISSUE THREAD

**Issue by GoogleCodeExporter** [liudonghua123/ie7-js#44]
```
Can we have a way to exclude stylesheets from ie7's scope?
```

Original issue reported on code.google.com by `makin...@gmail.com` on 25 Feb 2008 at 9:50

**Comment by GoogleCodeExporter** [comment]
```
Good idea.
```

Original comment by `dean.edw...@gmail.com` on 25 Feb 2008 at 11:40
- Changed state: **Accepted**
- Added labels: **Priority-High**, **Type-Enhancement**
- Removed labels: **Priority-Medium**, **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
This will probably be enabled by inclusion of a CSS comment:

/* _IE7_IGNORE_ */

Any objections?
```

Original comment by `dean.edw...@gmail.com` on 26 Feb 2008 at 10:12

**Comment by GoogleCodeExporter** [comment]
```
Nope! Probably the most unobtrusive way to do it.
```

Original comment by `makin...@gmail.com` on 27 Feb 2008 at 12:13

**Comment by GoogleCodeExporter** [comment]
```
However, will ie7-js just ignore everything after the comment or ignore an 
entire
document with the comment? If its the later, I think it'd be smarter to let 
ie7-js
read anything before the comment and anything after the comment is ignored. Its 
not
completely necessary, but its certainly a nice-to-have.
```

Original comment by `makin...@gmail.com` on 27 Feb 2008 at 12:17

**Comment by GoogleCodeExporter** [comment]
```
You're right. Ignoring everything after the comment is better.
```

Original comment by `dean.edw...@gmail.com` on 27 Feb 2008 at 1:20

**Comment by GoogleCodeExporter** [comment]
```
What about:
/* START_IE7_IGNORE_ */
<link rel=stylesheet />
/* END_IE7_IGNORE_ */
```

Original comment by `drew.fre...@gmail.com` on 2 Mar 2008 at 11:03

**Comment by GoogleCodeExporter** [comment]
```
Is this the same issue as ie7.js not respecting media queries such as 
media="only screen and (max-device-
width: 480px)"?
```

Original comment by `carise...@gmail.com` on 13 Jun 2008 at 12:09

**Comment by GoogleCodeExporter** [comment]
```
http://banjax.com/t/ie7-js/with.html
http://banjax.com/t/ie7-js/without.html

These may help, then again they may prove I'm an idiot.

With IE7-js IE5.5-7 will parse all the stylesheets (IE8 acts properly and 
ignores the last two); without IE7-js 
IE5.5-8 will parse only the first (screen) stylesheet, which is correct afaik.


```

Original comment by `carise...@gmail.com` on 13 Jun 2008 at 1:14

**Comment by GoogleCodeExporter** [comment]
```
Does anybody have a solution for this yet? I am running into problems with IE7 
and
dojo style sheets. The dojo styles work fine without the patch, loading the 
patch
breaks them. Unfortunately I can't live without either library!

If there is any way I can make IE7 skip parsing either specific css files or 
specific
nodes I would love to hear about it.
```

Original comment by `alerque` on 28 Jul 2008 at 11:25

**Comment by GoogleCodeExporter** [comment]
```
MAybe Im missing something here, but why not use conditional comments to 
exclude ie7?
http://msdn.microsoft.com/workshop/author/dhtml/overview/ccomment_ovw.asp
```

Original comment by `thetoolman` on 3 Sep 2008 at 8:27

**Comment by GoogleCodeExporter** [comment]
```
thetoolman: Your link to msdn is broken, but I don't think it matters really. 
The
problem is not excluding IE7 (the browser) from parsing either the patch or some
styles, the problem is in getting IE7 (the patch) to not modify the css 
declarations
on some parts of documents. Conditional comments are no use for this because 
they
only stop the browser from processing or not processing a block of something. 
We want
the patch to parse some things but not others and there is no provision for 
that. You
will notice some of the suggestions above are for what are basically conditional
comments but targeted at IE7 (the patch).
```

Original comment by `alerque` on 3 Sep 2008 at 8:41

**Comment by GoogleCodeExporter** [comment]
```
Any updates with this one?

Also, does it really need to be as complex a string as that?
Why not something simpler, such as /* !IE7 */


```

Original comment by `Hunn...@gmail.com` on 6 Nov 2008 at 7:01

**Comment by GoogleCodeExporter** [comment]
```
This is necessity, I use ExtJS, it contains huge css files, so ie7-js takes a 
lot of
time to parse it, ie just hangs for a few seconds. I would love to exclude it 
somehow
from ie7-js parsing.
```

Original comment by `mirko%m-...@gtempaccount.com` on 23 Mar 2009 at 4:27

**Comment by GoogleCodeExporter** [comment]
```
It been implemented?
Can be in two modes:
(how cited by drew.freyling with modf)
/*<IEonly>*/
 {...}
*/</IEonly>*/
Will processed ONLY if is IE (useful to CSSHacks)

Or can be "inline":
{...} /*IEonly*/
{...}
{...} /*IEonly*/

Bye.
```

Original comment by `david7...@gmail.com` on 2 May 2009 at 3:43

**Comment by GoogleCodeExporter** [comment]
```
@david.... english ? do you speak it ?

This feature would be awesome, currently I'm having issues with ie8.js and 
YUI's reset-fonts-grids
```

Original comment by `jaysw...@gmail.com` on 15 May 2009 at 8:52

**Comment by GoogleCodeExporter** [comment]
```
Any progress on this? It would extremely helpful!
```

Original comment by `cultosau...@gmail.com` on 4 Jun 2009 at 12:43

**Comment by GoogleCodeExporter** [comment]
```
I've tried the examples listed on this page to fix a problem with using the YUI
library with IE9.js, but none seem to work, is there an example I can copy, or 
has
the functionality not been implemented yet?

Thanks and regards

Andy.
```

Original comment by `andrew.k...@gmail.com` on 19 Apr 2010 at 8:46

**Comment by GoogleCodeExporter** [comment]
```
Please explain how to use this? I have tried to do it, but it doesn't work at 
all.
```

Original comment by `jariwala...@gmail.com` on 16 Sep 2010 at 1:27

**Comment by GoogleCodeExporter** [comment]
```
That is of high importance and priority as the lack of it makes most of CSS 
frameworks and reset style sheets ineffective. Any "inherit" rules are ignored 
and break the design.

As proposed already, it should allow to exclude a range of declaration in, 
preferably, standard way, ie. by a comment. The simpler, the better:

```

border: none;
/\* ie7_ignore_start _/
font-family: inherit;
font-size: inherit;
/_ ie7_ignore_end */
font-style: inherit;
font-weight: inherit;

```

This should also allow to exclude the whole style sheet with an attribute to 
"link" element, eg.:

```

<link href="style/screen.css" rel="stylesheet" type="text/css" media="screen, 
projection" ie7="false" />

```
```

Original comment by `grzegorz...@gmail.com` on 18 Oct 2010 at 1:59

**Comment by GoogleCodeExporter** [comment]
```
Hi ! 
What happened to this feature ? I need it on one of my company website because 
media queries are applied but without taken into account the specified rules
```

Original comment by `thomas.s...@gmail.com` on 26 Nov 2012 at 4:37
